include/dali/core/cuda_error.h

```diff
@@ -156,6 +162,11 @@ inline void CUDA_CALL(T status) {
   return dali::cudaResultCheck(status);
 }
 
+template <typename T>
+inline void CUDA_CALL_EX(T status, std::string extra) {
```

Comment:
It's a bad idea to make multiple copies of a `string` for the sake of an unlikely event (an error).
```suggestion
/**
 * @param status A status code that indicates an error or success
 * @param extra Additional information about an error, if it occurs. The
 *              caller should take care not to construct a string solely for the
 *              purpose of this parameter due to performance overhead.
 */
inline void CUDA_CALL_EX(T status, const std::string &extra) {
```